lib/gauge_force.cu

```diff
@@ -47,20 +51,35 @@ namespace quda {
     checkLocation(mom, u);
     if (mom.Reconstruct() != QUDA_RECONSTRUCT_10) errorQuda("Reconstruction type %d not supported", mom.Reconstruct());
 
-    // create path struct in a single allocation
-    size_t bytes = 4 * num_paths * path_max_length * sizeof(int) + num_paths*sizeof(int) + num_paths*sizeof(double);
-    void *buffer = pool_device_malloc(bytes);
-    paths p(buffer, bytes, input_path, length_h, path_coeff_h, num_paths, path_max_length);
+    paths p(input_path, length_h, path_coeff_h, num_paths, path_max_length);
+
+    // gauge field must be passed as first argument so we peel off its reconstruct type
+    instantiate<GaugeForce_,ReconstructNo12>(u, mom, epsilon, p);
+    p.free();
```

Comment:
Can the `free` be enforced in the destructor?

Reply:
No, otherwise the copy made by the driver that is made when calling the kernel would free the buffer when it goes out of scope.